### PR TITLE
WIP: Library update for integration with other operators

### PR DESCRIPTION
**Pull request by chambridge**
* Provide an out of the box summarizer
* Remove jitter aspects
* Handle sending multiple files or single file

### CODE REVIEW

File: insights/insightsuploader/insightsuploader.go [hunk 1]
```diff
@@ -75,110 +102,60 @@ func (c *Controller) Run(ctx context.Context) {
...
-				Type:     c.client.GetMimeType(),
-				Contents: source,
-			}); err != nil {
-				klog.V(2).Infof("Unable to upload report after %s: %v", time.Now().Sub(start).Truncate(time.Second/100), err)
-				versionError := err == insightsclient.ErrWaitingForVersion || err == insightsclient.ErrObtainingForVersion
-				if versionError {
-					initialDelay = wait.Jitter(interval/8, 1) - interval/8
-					if c.reporter.SafeInitialStart() {
-						initialDelay = wait.Jitter(time.Second*15, 1)
+			sourceIndex := 0
```
Comment (tremes) [inline insights/insightsuploader/insightsuploader.go:119]: What is the intention of this please?

File: insights/summarize/basicdisksummarizer.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	if err != nil {
+		return nil, false, nil
+	}
+	klog.V(4).Infof("Found files to send: %v", summaryFile)
+	f, err := os.Open(filepath.Join(b.basePath, summaryFile))
+	if err != nil {
+		return nil, false, nil
+	}
+	return f, true, nil
+}
```
Comment (tremes) [inline insights/summarize/basicdisksummarizer.go:105]: I originally thought a lib user would provide the file he/she wants to send/upload.

**Comment by tremes** [comment]
What is the plan to use the library please? Putting simply - user will create a new insightsclient, pass it to insightsuploader's new function and then will call run for the insightsuploader?

Do we want to run the uploader here periodically or should it be caller's responsibility? If we run it periodically then it will probably make sense to make the period configurable.

File: insights/insightsuploader/insightsuploader.go (hunk 1 above)
Comment (chambridge) [reply insights/insightsuploader/insightsuploader.go:119]: Code currently only supports sending the "latest" package in a directory. However if the contents is too large it needs to be split across multiple payloads. This allows for all items in the directory to be processed.

File: insights/summarize/basicdisksummarizer.go (hunk 2 above)
Comment (chambridge) [reply insights/summarize/basicdisksummarizer.go:105]: Library currently expects a Summarizer interface. We can change all that to just be give the full path to the file you want to upload. I'm fine with that; was trying to find a middle ground that provided something that I thought most every library user would need if they needed an implementation of the summarizer.

**Comment by chambridge** [comment]
@tremes Yes, trying to work through that with this work. Currently the plan would be:
- Create insightsclient
- Pass to insightsuploader New
- Call Run method

I had removed the periodic call and put the responsibility with the caller.

Thanks for your comments happy to make changes. If you have opinions on making this a simpler flow I'm all for that.  I would certainly prefer to just pass in a file (either the path or contents/io.ReadCloser) and the statusReporter provide details on the single payload upload.

File: insights/insightsuploader/insightsuploader.go (hunk 1 above)
Comment (tremes) [reply insights/insightsuploader/insightsuploader.go:119]: OK thanks

File: insights/summarize/basicdisksummarizer.go (hunk 2 above)
Comment (tremes) [reply insights/summarize/basicdisksummarizer.go:105]: Yeah I wanted to open this topic. I am not sure we need the summarizer TBH... I guess we will see once it's used somewhere (e.g the Cost operator)

**Comment by tremes** [comment]
@chambridge OK thanks. Well I meant this call https://github.com/RedHatInsights/insights-ingress-http-client/pull/11/files#diff-a1ed3778a4beb9ee464db48ef59bd2a6f2f18fc9eccabfee3c20ebc1e2ac4f8bL97 so I guess the "insightsuploader run" will run periodically :)) then it probably makes sense to keep the summarizer. 
Second option is to simply call `Upload` method and leave the timing aspect up to the caller. I guess this wouldn't require the summarizer and the statusReporter...I am not sure what's better....I need to think about it more :)

**Comment by chambridge** [comment]
@tremes Would something like the following track ith your second option?
https://github.com/RedHatInsights/insights-ingress-http-client/pull/13

**Comment by chambridge** [comment]
Closing in favor or #13.

**Pull request closed without merging**